### PR TITLE
Validate access to peep_loading_positions

### PR DESCRIPTION
**Pull request by janisozaur**
[21a313b7-e260-4469-9b08-c2b44e9ca38d-export.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/490647/21a313b7-e260-4469-9b08-c2b44e9ca38d-export.sv6.zip)

### CODE REVIEW

**Comment by IntelOrca** [comment]
While its good to assert, it doesn't fix the problem. Any ideas?

**Comment by janisozaur** [comment]
Would this be caused by "unlimited length trains" cheat?

**Comment by IntelOrca** [comment]
Not sure, a vehicle represents one single car of a train - so I can't quite see the connection.

**Comment by janisozaur** [comment]
Even though I made CIs green, it would be worth checking it properly if it hasn't broken any object loading.

File: src/openrct2/ride/vehicle.h [hunk 1]
```diff
@@ -77,8 +77,11 @@ typedef struct rct_ride_entry_vehicle {
 	uint8 draw_order;
 	uint8 special_frames;			// 0x60 , 0x7A
 	sint8* peep_loading_positions;	// 0x61 , 0x7B
+#ifdef NO_RCT2
+	uint16 peep_loading_positions_count;
+#endif
 } rct_ride_entry_vehicle;
-#ifdef PLATFORM_32BIT
+#if defined(PLATFORM_32BIT) && !defined(NO_RCT2)
```
Comment (Gymnasiast) [inline src/openrct2/ride/vehicle.h]: Could you also check for  `#if defined(PLATFORM_32BIT) && defined(NO_RCT2)` with the new struct size?

**Pull request merged**